runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -297,33 +299,54 @@ function LanguageTree:_get_injections()
 
     for pattern, match, metadata in self._injection_query:iter_matches(root_node, self._source, start_line, end_line+1) do
       local lang = nil
-      local injection_node = nil
+      local ranges = {}
       local combined = false
 
+      -- Directives can configure how injections are captured as well as actual node captures.
+      -- This allows more advanced processing for determining ranges and language resolution.
+      if metadata.content then
+        local content = metadata.content
+
+        -- Allow for captured nodes to be used
+        if type(content) == "number" then
+          content = {match[content]}
+        end
+
+        if content then
+          vim.list_extend(ranges, content)
+        end
+      end
+
+      if metadata.language then
+        lang = metadata.language
+      end
+
+      if metadata.combined ~= nil then
+        combined = metadata.combined
+      end
```

Comment:
If `metadate.combined` is `nil` or `false`, this'll be the same when using it in `if`s so no need to check here.